server/sonar-server/src/main/java/org/sonar/server/computation/monitoring/CEQueueStatusImpl.java

```diff
@@ -21,86 +21,53 @@
 
 import java.util.concurrent.atomic.AtomicLong;
 import org.sonar.ce.monitoring.CEQueueStatus;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.ce.CeQueueDto;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Preconditions.checkState;
 
 public class CEQueueStatusImpl implements CEQueueStatus {
-  private static final long PENDING_INITIAL_VALUE = Long.MIN_VALUE;
 
-  private final AtomicLong received = new AtomicLong(0);
-  private final AtomicLong pending = new AtomicLong(PENDING_INITIAL_VALUE);
+  private final DbClient dbClient;
   private final AtomicLong inProgress = new AtomicLong(0);
   private final AtomicLong error = new AtomicLong(0);
   private final AtomicLong success = new AtomicLong(0);
   private final AtomicLong processingTime = new AtomicLong(0);
 
-  @Override
-  public long initPendingCount(long initialPendingCount) {
-    checkArgument(initialPendingCount >= 0, "Initial pending count must be >= 0");
-    checkState(
-      pending.compareAndSet(PENDING_INITIAL_VALUE, initialPendingCount),
-      "Method initPendingCount must be used before any other method and can not be called twice");
-    return initialPendingCount;
-  }
-
-  @Override
-  public long addReceived() {
```

Comment:
is the received metric good all together?
I think it is a valuable metric, otherwise we can't track the activity of the task submission at all anymore
obviously it should be on the WebServer side only now